rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,108 @@
+but which has no active days of the week. The typicality of the service is tagged with a new value
+so it can be programmatically identified.
+
+All rail routes have two or more trips on this service, corresponding to the "canonical" patterns of
+service in each direction for any branches, as understood by stakeholders. Each such trip has a
+route pattern, for which it is the representative trip, and a shape, and the route pattern is
+considered "typical" for the route. If current service has trips following the same route pattern,
+they will use the canonical route pattern as well.
+
+In short, for consumers of the feed, a route pattern has typicality of `0` if and only if it visits
+a set of canonical stops in a canonical order.
+
+# Reference-level explanation
+
+The canonical service appears in the following GTFS files in the following way:
+
+- **calendar.txt** includes a new `service_id` to envelop all of the canonical trips on all the
+  routes. `start_date` and `end_date` should reflect the feed's valid start/end dates. All other
+  fields should have a value of `0`.
+- **calendar_attributes.txt** includes the new `service_id`. `service_schedule_type` should be
+  `"Other"`, and `rating_start_date` and `rating_end_date` should reflect the feed's valid start/end
+  dates. `service_schedule_typicality` should be `6`, and the
+  [docs](https://github.com/mbta/gtfs-documentation/blob/master/reference/gtfs.md) should be updated
+  with this new kind of value.
+- **route_patterns.txt** includes new route patterns for all routes/directions we want to hardcode.
+  `route_pattern_typicality` should be `1`, and `route_pattern_sort_order` should be a less than the
+  value for all other route patterns for that given route/direction. For routes with branches,
+  you'll need a pattern for each route/direction/branch (eg, for Providence/Stoughton Line, you
+  should have two new route patterns in the outbound direction).
+- **shapes.txt** can probably use the existing shapes.
+- **trips.txt** includes one new trip for each new route pattern. It should reference aforementioned
+  route pattern and shapes. It should reference the new `service_id` created above. Each of these
+  trips should be the `representative_trip_id` for their respective route patterns.
+- **stop_times.txt** includes all stops that trains in the route/direction passes through, even if
+  they are temporarily closed. Increment the arrival/departure times by one minute for each
+  subsequent stop. For a sample, see the attached GTFS file, the output of [✓ 🧪 🚝 Canonical
+  trips/patterns experiment](https://app.asana.com/0/881264583703207/1200210504369250), which
+  trialled this for Green Line D and Lowell Line (only in the outbound direction).
+
+This will be done as a new "step" of the Makefile called `canonical`, which runs between `truncate`
+and `current`. There will be a new a directory, `input/canonical/` to store any static data the step
+uses.
```

Comment:
I'd be interested to hear (perhaps in the "Rationale") why this method would be preferred over introducing the "canonical" schedule early in the process, and having subsequent steps that would normally create route patterns check whether they're duplicating a canonical one.